logging/logging.go

```diff
@@ -110,3 +110,31 @@ func NewObservedTestLogger(tb testing.TB) (Logger, *observer.ObservedLogs) {
 
 	return logger, observedLogs
 }
+
+type MemLogger struct {
+	Logger
+
+	tb       testing.TB
+	observer *observer.ObservedLogs
+}
+
+func (memLogger *MemLogger) OutputLogs() {
+	appender := NewTestAppender(memLogger.tb)
+	for _, loggedEntry := range memLogger.observer.All() {
+		appender.Write(loggedEntry.Entry, loggedEntry.Context)
+	}
+}
+
+func NewInMemoryLogger(tb testing.TB) *MemLogger {
```

Comment:
Surprised linter doesn't complain about lack of export comments here.

Reply:
You're right! I'm so confused by how our linter behaves.

Reply:
And this time it did complain:
![image](https://github.com/viamrobotics/rdk/assets/71458/e0d46cdb-f8dd-430c-b475-3b6976f6d85d)